ang/test/civicase-base/services/tags-helper.service.spec.js

```diff
@@ -0,0 +1,101 @@
+/* eslint-env jasmine */
+
+(() => {
+  describe('Tags Helper Service', () => {
+    let TagsHelper, TagsMockData;
+
+    beforeEach(module('civicase', 'civicase.data'));
+
+    beforeEach(inject((_TagsHelper_, _TagsMockData_) => {
+      TagsHelper = _TagsHelper_;
+      TagsMockData = _TagsMockData_.get();
+    }));
+
+    describe('when displaying tags', () => {
+      let tagMarkup;
+
+      beforeEach(() => {
+        tagMarkup = TagsHelper.formatTags({
+          text: 'Tag Name',
+          indentationLevel: '1'
+        });
+      });
+
+      it('returns all the case statuses', () => {
+        expect(tagMarkup).toEqual('<span style="margin-left:4px">Tag Name</span>');
+      });
+    });
+
+    describe('when displaying generic tags', () => {
+      let preparedTags, tags;
+
+      beforeEach(() => {
+        tags = [TagsMockData[0], TagsMockData[18]];
+        preparedTags = TagsHelper.prepareGenericTags(tags);
+      });
+
+      it('shows the tags hierarchy', () => {
+        expect(preparedTags).toEqual([{
+          id: '1',
+          name: 'Non-profit',
+          text: 'Non-profit',
+          description: 'Any not-for-profit organization.',
+          is_selectable: '1',
+          is_reserved: '0',
+          is_tagset: '0',
+          used_for: 'civicrm_contact',
+          indentationLevel: 0
+        }, {
+          id: '18',
+          name: 'L3',
+          text: 'L3',
+          parent_id: '1',
+          is_selectable: '1',
+          is_reserved: '0',
+          is_tagset: '0',
+          used_for: 'civicrm_activity',
+          created_id: '202',
+          created_date: '2018-12-27 07:38:15',
+          indentationLevel: 1
+        }]);
```

Comment:
Can we refactor it like this:

```js
      let excpectedTags;

      beforeEach(() => {
        tags = _.sample(TagsMockData, 2);

        expectedTags = _.cloneDeep(tags);
        expectedTags[0].parent_id = null;
        expectedTags[0].indentationLevel = 0;
        expectedTags[1].parent_id = expectedTags[1].id;
        expectedTags[1].indentationLevel = 1;

        preparedTags = TagsHelper.prepareGenericTags(tags);
      });

      it('shows the tags hierarchy', () => {
        expect(preparedTags).toEqual(expectedTags);
      });
```

I believe this is better because:

* We can update the tags mock data without having to worry about breaking tests.
* We explain both what is required (the second tag should be a child of the first one) and also what are the expected properties (`indentationLevel`) without having to compare this big object with the original values. For example, I'm not even sure if what changed is the `created_date`